### DISCUSSION TITLE
Change default build behavior in the Makefile

### DISCUSSION THREAD

**Discussion by kellinm** [kanidm/kanidm#772] [Ideas]
A question orthogonally related to https://github.com/kanidm/kanidm/issues/768

From my perspective, discovering that `ARGS` defaulted to an unreachable resource was a somewhat unexpected behavior.

@Firstyear and @yaleman - after https://github.com/kanidm/kanidm/pull/771 is finished up, would you be amenable to a merge request that does the following?

1. Sets `ARGS` to a default of empty string
1. Adds an example to the documentation with the current default value as something that could be passed using the `ARGS` environment variable

I wasn't sure if this should be an issue or a comment on https://github.com/kanidm/kanidm/issues/768, so instead I opted to make it a discussion so it's not cluttering things up in the issue tracker if it's not an acceptable behavior change.

**Comment by yaleman** [comment]
I'm down with it, @Firstyear's the only one likely affected 🤣

**Reply by Firstyear** [reply]
As mentioned on #768 , lets remove my hardcoded defaults so they work for you :)

**Comment by kellinm** [comment]
Continuing our conversation from #771  @Firstyear 

> I think the Makefile was more intended for "kind of" releases rather than development, but I guess we can take it that way and tidy it up as both a release and dev tool.

So, the reason I started poking at the Makefile and build documentation was a missing container for `1.1.0-alpha.8` on DockerHub.

I have been working through setting up identity services for a small community and, when I watched your recorded presentation from DevConf.cz, it really resonated with me and all the troubles I'd run into trying to get a modern authentication setup working.

Forgive me if I'm misunderstanding, but it sounds like the project is still figuring out the build and release process. My background is build, release, and distribution engineering so - if you need help in that area - I am glad to make myself useful and help with those tasks.

Let me know if you already have a clear vision that just needs hands to implement or if you'd entertain a proposal after a brief discussion to gain more insight about requirements and goals.

**Reply by Firstyear** [reply]
Yeah, I'd love some help with that! So far the release process is "manual-ish" which you can see in project_docs/RELEASE_CHECKLIST.md

I think the makefile was originally just to help me automate those releases, but to also do "devel" releases for people who wanted to run bleeding edge versions. That's why it has some hardcoded bits that relate to my setup to make it faster - such as the redis cache, and the mirrors that are more local to Australia.

But certainly, I'd love to hear what you think, and how we could do this better, we could do some "trial releases" or something to explore ideas and suggestions you may have. But also, I think so far, there are only really two of us that mostly develop the project, so even improvements to the development workflows are good to. A big goal is "keep it simple" you should be able to git clone; cargo test; But there are still rough edges and we can always improve them.

So yes, would love to hear more about what you think :)

**Reply by kellinm** [reply]
I agree with you :100: about ease of use.

Tooling is only helpful if people want to use it, otherwise it's just more technical debt!

I'll draft something up over the next week and tag you on an issue with a proposal.

**Reply by Firstyear** [reply]
I could not agree more with this. Even for such a new project we have built a little bit of that techdebt, but I've been spending time to pay back that loan recently. So that proposal sounds so good, I can't wait to see it. Feel free to PR it to the designs folder too, I think it's great to have them committed in the repo. Thanks!

**Reply by Firstyear** [reply]
PS: You'll be happy to know the current feature I'm working on is user-onboarding and credential reset links which is a big thing if you want something for community :)
